src/components/InputField/InputField.tsx

```diff
@@ -0,0 +1,45 @@
+import { useState, useCallback, useEffect } from 'react';
+import debounce from 'lodash.debounce';
+import { Person } from '../../types/Person';
+
+type Props = {
+  onQuery: (query: string) => void;
+  delay: number;
+  selectedPerson: Person | null;
+};
+
+export const InputField: React.FC<Props> = ({
+  delay,
+  onQuery,
+  selectedPerson,
+}) => {
+  const [query, setQuery] = useState('');
+
+  const applyQuery = useCallback(
+    debounce(onQuery, delay),
+    [],
```

Comment:
The dependencies array of useCallback is empty. This means the debounce function will not update if the props delay or onQuery change. You should include delay and onQuery in the array.